core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -653,6 +684,8 @@ public int hashCode() {
         result = 31 * result + (currentNodeId != null ? currentNodeId.hashCode() : 0);
         result = 31 * result + (relocatingNodeId != null ? relocatingNodeId.hashCode() : 0);
         result = 31 * result + (primary ? 1 : 0);
+        result = 31 * result + (int) (primaryTerm ^ (primaryTerm >>> 32));
+        ;
         result = 31 * result + (state != null ? state.hashCode() : 0);
         result = 31 * result + (int) (version ^ (version >>> 32));
```

Comment:
While not part of your change, here [too](https://github.com/elastic/elasticsearch/pull/14062/files#r42509768), for consistency?